packages/shared/src/lib/cores.ts

```diff
@@ -1,10 +1,22 @@
+import { isNullOrUndefined } from './func';
 import type { LoggedUser } from './user';
 import { CoresRole } from './user';
 
 export const verifyCoresRole = (
   user: LoggedUser,
   coresRole: CoresRole,
-): boolean => user?.coresRole === coresRole;
+): boolean => {
+  if (isNullOrUndefined(user?.coresRole)) {
+    return false;
+  }
 
-export const hasAccessToCores = (user: LoggedUser): boolean =>
-  !verifyCoresRole(user, CoresRole.None);
+  return user?.coresRole === coresRole;
+};
+
+export function hasAccessToCores(user: LoggedUser): boolean {
+  if (isNullOrUndefined(user?.coresRole)) {
+    return false;
+  }
```

Comment:
verifyCoresRole already takes care of it no?

Reply:
> Yeah let's do that, seems most optimized

I removed the `isNullOrUndefined` check in `verifyCoresRole` as you said, the `user?.coresRole === coresRole` already takes care of it.

Reply:
perfect 👌